eitprocessing/mixins/slicing.py

```diff
@@ -10,33 +10,29 @@
 
 
 class SelectByIndex(ABC):
-    time: NDArray
+    label: str
 
     def select_by_index(  # pylint: disable=too-many-arguments
         self,
         start: int | None = None,
         end: int | None = None,
-        start_inclusive: bool = True,
-        end_inclusive: bool = False,
         label: str | None = None,
     ) -> Self:
-        if not any((start, end)):
+        if start is None and end is None:
             warnings.warn("No starting or end timepoint was selected.")
             return self
 
-        self._check_time_sorted()
-
-        if start is None:
-            start = 0
-
+        start = start or 0
         if end is None:
-            end = len(self.time)
-
-        if not start_inclusive:
-            start += 1
+            end = len(self)
 
-        if end_inclusive:
-            end += 1
+        if label is None:
+            if start > end:
+                label = f"No frames selected from <{self.label}>"
+            elif start < end - 1:
+                label = f"Slice ({start}-{end-1}) of <{self.label}>"
```

Comment:
I would suggest using "Frames" instead of "Slice".

```suggestion
                label = f"Frames ({start}-{end-1}) of <{self.label}>"
```